### ISSUE TITLE
URL::action() does not accept "callable" array syntax

### ISSUE THREAD

**Issue by aaroncadrian** [laravel/framework#24770]
- Laravel Version: 5.6.26
- PHP Version: 7.2
- Database Driver & Version: MySQL 5.7

### Description:

There is a discrepancy between the URL Generation documentation and the actual URL Generator. 
 According to the [documentation](https://laravel.com/docs/5.6/urls#urls-for-controller-actions):

> You may also reference actions with a "callable" array syntax:
> ```
> use App\Http\Controllers\HomeController;
> 
> $url = action([HomeController::class, 'index']);
> ```

However, doing so throws this exception:

> ErrorException (E_WARNING)
> strpos() expects parameter 1 to be string, array given

The issue is coming from `Illuminate\Routing\UrlGenerator@formatAction` and `Illuminate\Routing\UrlGenerator@action` since there does not appear to be any logic that processes an array as the first argument to the `action` method.

Either the documentation has an error or the code has an error.

### Steps To Reproduce:

1. Inside a fresh Laravel project, open `routes/web.php`
2. Paste the following code:
    ```
    Route::get('/example', function() {
        return action([\App\Http\Controllers\HomeController::class, 'index']);
    });
    ```
3. Visit `http://{YOUR_PROJECT_URL}/example`

**Comment by devcircus** [comment]
Looks like the pr, #24738 for this functionality, was merged into 5.7, but the pr for the documentation [4361](https://github.com/laravel/docs/pull/4361), was merged into 5.6. Maybe @sebastiandedeyne can look into this.

**Comment by themsaid** [comment]
The docs were corrected, closing this then.
